kvstore/simplemap/simplemap.go

```diff
@@ -63,8 +63,8 @@ func (sm *simpleMap) Delete(key []byte) error {
 }
 
 // Len returns the number of key-value pairs in the store.
-func (sm *simpleMap) Len() int {
-	return len(sm.m)
+func (sm *simpleMap) Len() (int, error) {
+	return len(sm.m), nil
```

Comment:
To jump on my comment earlier, I think having length return an error feels strange and removing the iteration in pebble will help with this, and make it faster - in pebbles case constant time at least.

Reply:
We're querying the length from the database - which is the source of truth for the **data** - this is not different from any other query. I'd say we should treat it as an SQL query `COUNT(*)` - and as that operation can fail for any reason, we should return an error.